client/Wayland/wlf_input.c

```diff
@@ -171,9 +173,12 @@ BOOL wlf_handle_pointer_axis(freerdp* instance, const UwacPointerAxisEvent* ev)
 			return FALSE;
 	}
 
-	step = (uint32_t)abs(direction);
-	if (step > WheelRotationMask)
-		step = WheelRotationMask;
+	/* Wheel rotation steps:
+	 *
+	 * positive: 0 ... 0xFF  -> slow ... fast
+	 * negative: 0 ... 0xFF  -> fast ... slow
+	 */
+	step = (uint32_t)direction & 0xFF;
```

Comment:
This will not work as expected, for `ev->value == 256`, `step` should be `255` not `0`.